task_manager/task_manager/tasks/system_tasks.py

```diff
@@ -104,3 +129,73 @@ def get_task_specs(topic: str) -> TaskSpecs:
             task_server_type=TaskServerType.SERVICE,
             service_success_field="success",
         )
+
+
+class WaitTask(SystemTask):  # pylint: disable=too-few-public-methods
+    """Wait for a specified amount of time or until a cancel is called."""
+
+    def __init__(self, node: Node, topic: str):
+        self._node = node
+        self._topic = topic
+        self._server = ActionServer(
+            self._node,
+            Wait,
+            self._topic,
+            self._execute_cb,
+            cancel_callback=self._cancel_cb,
+            callback_group=MutuallyExclusiveCallbackGroup(),
+        )
+
+    def _execute_cb(self, goal_handle: ServerGoalHandle) -> Wait.Result:
+        """Callback for the Wait action server."""
+        duration_in_seconds = goal_handle.request.duration
+        start_time = self._node.get_clock().now()
+
+        # Wait indefinitely if duration is 0.0
+        if duration_in_seconds <= 0.0:
+            loop_time = 0.1
+            end_time = Time(nanoseconds=2**63 - 1, clock_type=ClockType.ROS_TIME)
+        else:
+            loop_time = 0.1 if duration_in_seconds > 0.1 else duration_in_seconds
+            end_time = start_time + Duration(nanoseconds=int(duration_in_seconds * 1e9))
+
+        feedback_frequency_ns = int(1.0 * 1e9)
+        last_feedback_stamp = start_time
+
+        while rclpy.ok() and self._node.get_clock().now() < end_time:
+            if goal_handle.is_cancel_requested:
+                goal_handle.canceled()
+                return Wait.Result()
+
+            if not goal_handle.is_active:
+                goal_handle.abort()
+                return Wait.Result()
+
+            # Publish feedback
+            if (self._node.get_clock().now() - last_feedback_stamp).nanoseconds >= feedback_frequency_ns:
+                last_feedback_stamp = self._node.get_clock().now()
+                goal_handle.publish_feedback(
+                    Wait.Feedback(remaining_time=float((end_time.nanoseconds - last_feedback_stamp.nanoseconds) / 1e9))
+                )
+
+            time.sleep(loop_time)
+
+        goal_handle.succeed()
+        return Wait.Result()
+
+    @staticmethod
+    def _cancel_cb(_goal_handle: ServerGoalHandle) -> CancelResponse:
+        return CancelResponse.ACCEPT
+
+    @staticmethod
+    def get_task_specs(topic: str) -> TaskSpecs:
+        return TaskSpecs(
+            task_name="system/wait",
+            blocking=True,
+            cancel_on_stop=True,
+            topic=topic,
+            cancel_reported_as_success=False,
```

Comment:
To me `cancel_reported_as_success=False` made sense when I was thinking about this. But I also see your point. Maybe in the indefinite case it makes more sense to be `True`. We lose a slight bit of information in the other case but don't know if that sliver of information would be useful.

Reply:
Yes, it is true that it will now always be incorrect in one of the two cases. I suppose the only way to prevent that would be to keep this property as False and then change the indefinite case in the callback: If we detect a cancel request but the wait time was indefinite, we would call `succeed()` instead of `cancel()` on the goal handle. This would lead to the correct final state of the action in all cases, in the indefinite case there would just be a short period in between where the action would be in a CANCELLING state.

Reply:
That sound good to me. Changed as suggested. Also fixed the tests which broke from yesterdays "quick last change before I finish the day" change...

Reply:
Great, looks good to me now!